microscopium/preprocess.py

```diff
@@ -549,39 +557,58 @@ def find_background_illumination(fns, radius=51, quantile=0.05,
     --------
     `correct_image_illumination`, `correct_multiimage_illumination`.
     """
-    # This function follows the "PyToolz" streaming data model to
-    # obtain the illumination estimate. First, define each processing
-    # step:
-    read = io.imread
+    read = tlz.partial(io.imread)
+
+    if input_bitdepth is None:
+        in_range = "image"
+    else:
+        in_range = (0, np.power(2, input_bitdepth) - 1)
+
+    if radius is None:
+        # get default radius from input image
+        im0 = mio.imread(fns[0])
+        radius = np.round(np.min(im0.shape) / 4).astype(np.uint16)
+        # ensure radius is odd
+        radius = radius - np.mod(radius, 2) + 1
+
+    rescale = tlz.partial(exposure.rescale_intensity, in_range=in_range)
```

Comment:
The [default](http://scikit-image.org/docs/dev/api/skimage.exposure.html#skimage.exposure.rescale_intensity) it based on the images' dtype, but sure I'll put that in explicitly. It'll make the code a bit easier to follow.